File: domain/src/test/scala/net/kemitix/thorp/domain/MD5HashData.scala
package net.kemitix.thorp.domain

object MD5HashData {

  object Root {
    val hash: MD5Hash  = MD5Hash("a3a6ac11a0eb577b81b3bb5c95cc8a6e")
    val base64: String = "o6asEaDrV3uBs7tclcyKbg=="
  }
  object Leaf {
    val hash: MD5Hash  = MD5Hash("208386a650bdec61cfcd7bd8dcb6b542")
    val base64: String = "IIOGplC97GHPzXvY3La1Qg=="
  }
  object BigFile {
    val hash: MD5Hash = MD5Hash("b1ab1f7680138e6db7309200584e35d8")
    object Part1 {
      val offset: Int   = 0
      val size: Int     = 1048576
      val hash: MD5Hash = MD5Hash("39d4a9c78b9cfddf6d241a201a4ab726")
    }
    object Part2 {
      val offset: Int   = 1048576
      val size: Int     = 1048576
      val hash: MD5Hash = MD5Hash("af5876f3a3bc6e66f4ae96bb93d8dae0")
    }
  }

}

File: domain/src/main/java/net/kemitix/thorp/domain/MD5HashData.java
package net.kemitix.thorp.domain;

public interface MD5HashData {

    class Root {
        public static final String hashString = "a3a6ac11a0eb577b81b3bb5c95cc8a6e";
        public static final MD5Hash hash = MD5Hash.create(hashString);
        public static final String base64 = "o6asEaDrV3uBs7tclcyKbg==";
        public static final RemoteKey remoteKey = RemoteKey.create("root-file");
        public static final Long size = 55L;
    }
    class Leaf {
        public static final String hashString = "208386a650bdec61cfcd7bd8dcb6b542";
        public static final MD5Hash hash = MD5Hash.create(hashString);
        public static final String base64 = "IIOGplC97GHPzXvY3La1Qg==";
        public static final RemoteKey remoteKey = RemoteKey.create("subdir/leaf-file");
        public static final Long size = 58L;
    }
    class BigFile {
        public static final String hashString = "b1ab1f7680138e6db7309200584e35d8";
        public static final MD5Hash hash = MD5Hash.create(hashString);

        public static class Part1 {
            public static final int offset = 0;
            public static final int size = 1048576;
            public static final String hashString = "39d4a9c78b9cfddf6d241a201a4ab726";
            public static final MD5Hash hash = MD5Hash.create(hashString);
        }
        public static class Part2 {
            public static final int offset = 1048576;
            public static final int size = 1048576;
            public static final String hashString = "af5876f3a3bc6e66f4ae96bb93d8dae0";
            public static final MD5Hash hash = MD5Hash.create(hashString);
        }
    }

}
